cmd/admin-handlers.go

```diff
@@ -1758,3 +1762,239 @@ func (a adminAPIHandlers) ServerHardwareInfoHandler(w http.ResponseWriter, r *ht
+				Version:      Version,
+				CommitID:     CommitID,
+				DeploymentID: globalDeploymentID,
+				SQSARN:       globalNotificationSys.GetARNList(),
+				Region:       globalServerRegion,
+			},
+		},
+	})
+
+	// Get CPU load details from local server's cpu(s)
+	cpu := getLocalCPULoad(globalEndpoints, r)
+	// Notify all other MinIO peers to report cpu load numbers
+	cpus := globalNotificationSys.CPULoadInfo()
+	cpus = append(cpus, cpu)
+
+	// Get mem usage details from local server(s)
+	m := getLocalMemUsage(globalEndpoints, r)
+	// Notify all other MinIO peers to report mem usage numbers
+	mems := globalNotificationSys.MemUsageInfo()
+	mems = append(mems, m)
+
+	// Fetch the Drive info
+	// Drive Perf is only implemented for Erasure coded backends
+	if !globalIsXL {
+		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrMethodNotAllowed), r.URL)
+		return
+	}
+
+	var size int64 = madmin.DefaultDrivePerfSize
+	// if sizeStr, found := vars["size"]; found {
+	// 	var err error
+	// 	if size, err = strconv.ParseInt(sizeStr, 10, 64); err != nil || size <= 0 {
+	// 		writeErrorResponseJSON(ctx, w, errorCodes.ToAPIErr(ErrBadRequest), r.URL)
+	// 		return
+	// 	}
+	// }
+	// Get drive performance details from local server's drive(s)
+	dp := getLocalDrivesPerf(globalEndpoints, size, r)
+
+	// Notify all other MinIO peers to report drive performance numbers
+	dps := globalNotificationSys.DrivePerfInfo(size)
+	dps = append(dps, dp)
+
+	// Construct server information
+	serversInfo := []madmin.ServerStruct{}
+
+	for i, sInfo := range serverInfo {
+		serverInfo := madmin.ServerStruct{}
+		//diskInfo := madmin.DiskStruct{}
+		if sInfo.Error != "" {
+			fmt.Println(" Error")
+		} else {
+			serverInfo.Endpoint = sInfo.Addr
+			serverInfo.Uptime = sInfo.Data.Properties.Uptime
+			serverInfo.Version = sInfo.Data.Properties.Version
+			serverInfo.CommitID = sInfo.Data.Properties.CommitID
+			serverInfo.CPU = madmin.ServerCPUInfo{
+				Cores:        " Test ",
+				Model:        " Intel ",
+				Load:         cpus[i].Load,
+				HistoricLoad: cpus[i].HistoricLoad,
+			}
+			serverInfo.Memory = madmin.ServerMemInfo{
+				Usage:         mems[i].Usage,
+				HistoricUsage: mems[i].HistoricUsage,
+			}
+			diskPerf := dps[i].Perf
+
+			for j, dp := range diskPerf {
+				serverInfo.Disks[j].Path = dp.Path
+				if dp.Error != "" {
+					serverInfo.Disks[j].State = "Offline"
+				} else {
+					if storageInfo.Backend.Type == BackendErasure {
+						serverInfo.Disks[j].State = "Online"
+						serverInfo.Disks[j].Model = "To do"
+						serverInfo.Disks[j].Totalspace = 10
+						serverInfo.Disks[j].Usedspace = 3
+						serverInfo.Disks[j].Readthroughput = dp.ReadSpeed
+						serverInfo.Disks[j].Writethroughput = dp.WriteSpeed
+						serverInfo.Disks[j].Readlatency = " To do"
+						serverInfo.Disks[j].Writelatency = " To do"
+						serverInfo.Disks[j].Utilization = float64(serverInfo.Disks[j].Usedspace/serverInfo.Disks[j].Totalspace) * 100
```

Comment:
Cast to float before dividing, this will probably not work and will probably always return 0.

Also, that will at least not crash if `serverInfo.Disks[j].Totalspace` for whatever reason should be 0.